src/core/lookup.c

```diff
@@ -746,8 +768,24 @@ QuicLookupAddLocalCid(
             Hash);
 
     if (ExistingConnection == NULL) {
-        Result =
-            QuicLookupInsertLocalCid(Lookup, Hash, SourceCid, TRUE);
+        QUIC_CID_HASH_ENTRY* CID =
```

Comment:
Just a thought, but we might need to start using pools for allocating and freeing all these entries so they don't hurt perf.